app/message_error_handler.py

```diff
@@ -0,0 +1,107 @@
+import base64
+import hashlib
+import logging
+
+import requests
+from pika import BasicProperties
+from structlog import wrap_logger
+
+from app.rabbit_context import RabbitContext
+from config import Config
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def _report_exception_for_advice(message_hash, service, queue, exception_class, exception_trace):
+    exception_report_payload = {
+        'messageHash': message_hash,
+        'service': service,
+        'queue': queue,
+        'exceptionClass': exception_class,
+        'exceptionMessage': exception_trace,
+    }
+
+    response = requests.post(f'{Config.EXCEPTION_MANAGER_URL}/reportexception', json=exception_report_payload)
+    response.raise_for_status()
+    return response.json()
+
+
+def _quarantine_message(body: bytes, message_hash, exception_class, headers):
+    quarantine_headers = headers or {}
+    quarantine_headers['quarantineOrigin'] = {
+        'exchange': Config.RABBIT_EXCHANGE,
+        'queue': Config.RABBIT_QUEUE,
+        'routing-keys': [Config.RABBIT_ROUTING_KEY]
+    }
+    _quarantine_message_in_exception_manager(body, message_hash, exception_class, quarantine_headers)
+    _quarantine_message_in_rabbit(body, quarantine_headers)
+
+
+def _quarantine_message_in_exception_manager(body: bytes, message_hash, exception_class, headers):
+    quarantine_payload = {
+        'messageHash': message_hash,
+        'messagePayload': base64.b64encode(body).decode(),
+        'service': Config.NAME,
+        'queue': Config.RABBIT_QUEUE,
+        'exceptionClass': exception_class,
+        'routingKey': Config.RABBIT_ROUTING_KEY,
+        'contentType': 'application/json',
+        'headers': headers,
+    }
+
+    response = requests.post(f'{Config.EXCEPTION_MANAGER_URL}/storeskippedmessage', json=quarantine_payload)
+    response.raise_for_status()
+
+
+def _quarantine_message_in_rabbit(body: bytes, headers):
+    properties = BasicProperties(content_type='application/json', headers=headers)
```

Comment:
of course, this doesn't anticipate the possibility that somebody sends a message _without_ the content type set, which would cause problems - this may mask such a problem

Reply:
It's now passing straight through the `properties` from the original message callback which should cover this. The consumer in Python doesn't actually check or care about the `content_type` property anyway.